src/main/flight/gps_rescue.c

```diff
@@ -395,25 +403,53 @@ void rescueAttainPosition()
 }
 
 // Very similar to maghold function on betaflight/cleanflight
-void setBearing(int16_t deg)
+void setBearing(int16_t desiredHeading)
 {
-    int16_t dif = DECIDEGREES_TO_DEGREES(attitude.values.yaw) - deg;
+    float errorAngle = DECIDEGREES_TO_DEGREES(attitude.values.yaw) - desiredHeading;
+
+    // Determine the most efficient direction to rotate
+    if (errorAngle <= -180) {
+        errorAngle += 360;
+    } else if (errorAngle > 180) {
+        errorAngle -= 360;
+    }
 
-    if (dif <= -180) {
-        dif += 360;
-    } else if (dif > 180) {
-        dif -= 360;
+    errorAngle *= -GET_DIRECTION(rcControlsConfig()->yaw_control_reversed);
+    const int errorAngleSign = (errorAngle < 0) ? -1 : 1;
+    const float errorAngleAbs = ABS(errorAngle);
+        
+    // Calculate a desired yaw rate based on a maximum limit beyond
+    // an error window and then scale the requested rate down inside
+    // the window as error approaches 0.
+    if (errorAngleAbs > GPS_RESCUE_RATE_SCALE_DEGREES) {
```

Comment:
```
rescueYaw = constrain(errorAngle / GPS_RESCUE_RATE_SCALE_DEGREES * GPS_RESCUE_MAX_YAW_RATE, -GPS_RESCUE_MAX_YAW_RATE, GPS_RESCUE_MAX_YAW_RATE);

Reply:
Yes, that will work. I'm going to wait to optimize it until after we get flight tests to confirm that this rate calculation logic gives practical results.

Reply:
Optimization is only secondary ... the idea is to make code easier to read ..